Framework: Remove sites-list from devdocs blocks

tyxla opened the pull request:
This PR removes the `sites-list` usage from devdocs, and deletes the no longer used `sites-list` example. Also, it adds `<Site />` and `<AllSites />` examples to the devdocs blocks page, as they were included in the deleted example and we don't want to lose them.

Testing:
* Verify http://calypso.localhost:3000/devdocs/blocks works properly.
* Verify the sites-list example is indeed no longer used anywhere.
* Verify http://calypso.localhost:3000/devdocs/blocks/all-sites works properly.
* Verify http://calypso.localhost:3000/devdocs/blocks/site works properly.

gwwar commented:
@tyxla this is an example for `<Site />` and `<AllSites />` which appear to still be in use. Unless these blocks are deprecated, can we update the example to be backed by redux data instead of removing it entirely?

tyxla commented:
@gwwar that's a reasonable comment, but I really prefer adding those as separate examples than keeping them together within a sites-list example. So, I've updated this PR to introduce separate examples for `<Site />` and `<AllSites />`, and added those to the devdocs blocks page. Does that sound good to you?

gwwar commented on client/my-sites/all-sites/docs/example.jsx:
```diff
@@ -0,0 +1,18 @@
+import AllSites from 'my-sites/all-sites';
+import Card from 'components/card';
+
+const AllSitesExample = () => (
```
I believe this one needs a displayName too.

tyxla replied:
Actually, it doesn't:

![](https://cldup.com/iIDBbUeQmE.png)

gwwar replied:
The issue is for the wpcalypso, since we mangle variable names in prod environments. https://wpcalypso.wordpress.com/devdocs/blocks/u

![screen shot 2017-05-25 at 9 41 10 am](https://cloud.githubusercontent.com/assets/1270189/26460390/5938b844-412e-11e7-846e-959eab4783e0.png)

tyxla replied:
Oops! Will take care of that

tyxla replied:
@gwwar addressed in #14480.

tyxla replied:
Thanks for catching it 🙇
